src/arc_utilities/algorithms.py

```diff
@@ -0,0 +1,20 @@
+import collections.abc
+
+
+def update(d, u):
```

Comment:
use more descriptive names for `d` and `u`

also `nested_update` may be a better name